### ISSUE TITLE
You really should make the access to optimizers and schedulers more comprehensible and more detailed.

### ISSUE THREAD

**Issue by onbigion13** [Lightning-AI/pytorch-lightning#20214]
### 📚 Documentation

I really appreciate that the method `configure_optimizers` . In your docs for `configure_optimizers`, you mentioned that one can return 6 types of result for this method. But you don't give any instruction about how to access a single optimizer or a single scheduler for corresponding result type. 

For example, in the docs for `configure_optimizers` ,  you gave an example return type:
```python 
 def configure_optimizers(self):
    optimizer1 = Adam(...)
    optimizer2 = SGD(...)
    scheduler1 = ReduceLROnPlateau(optimizer1, ...)
    scheduler2 = LambdaLR(optimizer2, ...)
    return (
        {
            "optimizer": optimizer1,
            "lr_scheduler": {
                "scheduler": scheduler1,
                "monitor": "metric_to_track",
            },
        },
        {"optimizer": optimizer2, "lr_scheduler": scheduler2},
    )
```

Then for this type of return, how should we access the optimizers and schedulers by `self.optimizers()` and `self.lr_schedulers()` method? 

This is significantly important for manual optimization and BP.

Hope you improve and perfect the configuration and access of sophisticated optimizers and lr_schedulers.

cc @borda
